src/RockBridge.cpp

```diff
@@ -1,193 +0,0 @@
-
-#include <base/typekit/Plugin.hpp>
-#include <base/transports/corba/TransportPlugin.hpp>
-#include <base/transports/typelib/TransportPlugin.hpp>
-#include <base/transports/mqueue/TransportPlugin.hpp>
-
-#include <gazebo/typekit/Plugin.hpp>
-#include <gazebo/transports/corba/TransportPlugin.hpp>
-#include <gazebo/transports/typelib/TransportPlugin.hpp>
-#include <gazebo/transports/mqueue/TransportPlugin.hpp>
-
-
-using namespace gazebo;
-
-//======================================================================================
-
-void RockBridge::Load(int _argc , char** _argv)
-{
-	RTT::corba::ApplicationServer::InitOrb(_argc, _argv);
-	
-	// Import typekits to allow RTT convert the types used by the components
-	RTT::types::TypekitRepository::Import(new orogen_typekits::stdTypekitPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::stdCorbaTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::stdMQueueTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::stdTypelibTransportPlugin);
-	
-	RTT::types::TypekitRepository::Import(new orogen_typekits::baseTypekitPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::baseCorbaTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::baseMQueueTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::baseTypelibTransportPlugin);
-	
-	RTT::types::TypekitRepository::Import(new orogen_typekits::gazeboTypekitPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::gazeboCorbaTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::gazeboMQueueTransportPlugin);
-	RTT::types::TypekitRepository::Import(new orogen_typekits::gazeboTypelibTransportPlugin);
-	
-		
-	// Each simulation step the Update method is called to update the simulated sensors and actuators
-	eventHandler.push_back(
-			event::Events::ConnectWorldUpdateBegin(
-					boost::bind(&RockBridge::updateBegin,this, _1)));
-	eventHandler.push_back(
-			event::Events::ConnectWorldUpdateEnd(
-					boost::bind(&RockBridge::updateEnd,this)));
-	eventHandler.push_back(
-			event::Events::ConnectWorldCreated(
-					boost::bind(&RockBridge::worldCreated,this, _1)));
-	eventHandler.push_back(
-			event::Events::ConnectAddEntity(
-					boost::bind(&RockBridge::modelAdded,this, _1)));
-}
-//======================================================================================
-// worldCreated() is called every time a world is added
-void RockBridge::worldCreated(std::string const& worldName)
-{
-    gzmsg << "RockBridge: initializing world: " << worldName << std::endl;
-
-	physics::WorldPtr world = physics::get_world(worldName);
-    if (!world)
-    {
-            gzerr << "RockBridge: cannot find world " << worldName << std::endl;
-            return;
-    }
-
-	int environment = GROUND;
-	std::string world_name("underwater");
-	if(world_name.compare((world)->GetName()) == 0)
```

Comment:
See [the related comment in my review of orogen/gazebo](https://github.com/Brazilian-Institute-of-Robotics/simulation-orogen-gazebo/pull/1/files#r24113134)

However, I thought it would be possible to add additional tags / attributes to the SDF file that you would access from there. If we really need to keep this mechanism, it would be a lot more reliable.